rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -867,7 +868,7 @@ public void visitArgumentlistExpression(ArgumentListExpression expression) {
                         if (omitParentheses == null) {
                             after = sourceBefore(")");
                         }
-                    } else {
+                    } else if (!(arg instanceof J.Lambda) || !(lastArgumentsAreAllClosures && omitParentheses != null)) {
```

Comment:
The negations here were slightly confusing to me as well; All tests pass if I apply this refactor; does that still fit what you were after?
![image](https://github.com/user-attachments/assets/1d3d9f84-f870-4ce0-8c87-43352c8a9658)
```suggestion
                    } else if (!(arg instanceof J.Lambda && lastArgumentsAreAllClosures && omitParentheses != null)) {
```

Reply:
It's logically the same, so it makes sense the test do succeed. In English this is what I want to convey:
>   `argument ` is a Closure where I know the method has parentheses and the method has trailing lambda's (that includes lambda's only), skip next part.
  
 Seems to me that's exactly what the refactor does also say, so let's go for it!